ydb/library/yql/dq/opt/dq_opt_join_cost_based.cpp

```diff
@@ -272,8 +273,17 @@ class TOptimizerNativeNew: public IOptimizerNew {
         TJoinHypergraph<TNodeSet> hypergraph = MakeJoinHypergraph<TNodeSet>(joinTree, hints);
         TDPHypSolver<TNodeSet> solver(hypergraph, this->Pctx);
 
-        if (solver.CountCC(MaxDPhypTableSize_) >= MaxDPhypTableSize_) {
+        if (solver.CountCC(MaxDPHypTableSize_) >= MaxDPHypTableSize_) {
             YQL_CLOG(TRACE, CoreDq) << "Maximum DPhyp threshold exceeded";
+            if (ExprCtx) {
+                ExprCtx->AddWarning(
+                    YqlIssue(
+                        {}, TIssuesIds::DQ_OPTIMIZE_ERROR, 
+                        "Cost Based Optimizer didn't work: "
```

Comment:
I don't like the text here, lets change it to:
"Cost Based Optimizer could not be applied to this query:"